root/etc/cont-init.d/20-drupal-files-dirs

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/with-contenv bash
+
+for DIR in $(ls -d ${DOCROOT}/sites/*/files); do
+  s6-echo -- Fixing ownership and permissions of $DIR
+  chown apache:apache $DIR
```

Comment:
In the case where the files directory is on my host laptop it does, but in that case the permissions don't matter as the web server process can successfully write to that directory and an install proceeds flawlessly. The condition where the permission change is needed is when the files directory has been placed on the data volume (or similarly mapped within the docker host).

IE, if the volumes for your web container look like:

```
    volumes:
      # Main code volume mount.
      - .:/var/www
      - /data/drupal8/files:/var/www/build/html/sites/default/files
```

In which case you'll get a permission error when you try to install because the files directory is not writable.